rbx_xml/src/types/mod.rs

```diff
@@ -99,6 +101,7 @@ macro_rules! declare_rbx_types {
         ) -> Result<(), EncodeError> {
             match value {
                 $(Variant::$variant_name(value) => value.write_outer_xml(xml_property_name, writer),)*
+                Variant::Attributes(value) => write_attributes(writer, xml_property_name, value),
```

Comment:
We should implement Attributes the same way that we implement other types. I don't think there's any reason to special-case them here. Look at how other types (not `Tags`!) are implemented, via `XmlType` and `write_outer_xml`.

Reply:
Combing back through this code, I remember why we had to implement Tags and Attributes this way, ugh. rbx_xml pre-dates most of the work we did on reflection to get rbx_binary functional, and so it wasn't built with types like Tags in mind.

Reply:
I saw you made another pull request for this. Should I remove my integration?